File: app/src/main/java/com/shuniuyun/material/activity/BottomSheetDialogActivity.java
package com.shuniuyun.material.activity;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetBehavior;
import com.google.android.material.bottomsheet.BottomSheetDialog;
import com.shuniuyun.material.R;

/**
 * author:wyb
 */
public class BottomSheetDialogActivity extends BaseActivity implements View.OnClickListener {
    private BottomSheetDialog mBottomSheetDialog;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bottomsheetdialog);
        createBottomSheetDialog();
        setBehaviorCallback();
    }

    private void createBottomSheetDialog() {
        mBottomSheetDialog = new BottomSheetDialog(this);
        View view = LayoutInflater.from(this).inflate(R.layout.view_bottomsheetdialog, null, false);
        mBottomSheetDialog.setContentView(view);
    }

    private void setBehaviorCallback() {
        View view = mBottomSheetDialog.getDelegate().findViewById(R.id.design_bottom_sheet);
        final BottomSheetBehavior bottomSheetBehavior = BottomSheetBehavior.from(view);
        bottomSheetBehavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {

            public void onStateChanged(@NonNull View bottomSheet, int newState) {
                if (newState == BottomSheetBehavior.STATE_HIDDEN) {
                    mBottomSheetDialog.dismiss();
                    bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
                }
            }

            public void onSlide(@NonNull View bottomSheet, float slideOffset) {}
        });
    }

    @Override
    public void onClick(View v) {
        if (mBottomSheetDialog.isShowing()) {
            mBottomSheetDialog.dismiss();
        } else {
            mBottomSheetDialog.show();
        }
    }

}

File: app/src/main/java/com/shuniuyun/material/activity/BottomSheetDialogActivity.kt
package com.shuniuyun.material.activity

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.shuniuyun.material.R
import com.shuniuyun.material.utils.SheetDialogFactory

/**
 * author:wyb
 * 弹出一个可上下拉伸的弹框，内容可自定义
 */
class BottomSheetDialogActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var fullSheetDialog: SheetDialogFactory.FullSheetDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bottomsheetdialog)
        initView()
    }

    private fun initView() {
//        SheetDialogFactory.initialize(this, R.layout.view_bottomsheetdialog)
        fullSheetDialog = SheetDialogFactory.create(this, R.layout.view_bottomsheetdialog)
    }

    override fun onClick(v: View) {
        if (!fullSheetDialog.isShowing) fullSheetDialog.show()
    }

}